app/controllers/discussions_controller.rb

```diff
@@ -48,14 +51,25 @@ def update
 
   private
 
+    def check_discussion_is_upcoming
+      if !(@discussion.date.to_s > Date.current.to_s)
```

Comment:
date = Date.new(2007, 11, 10)       # => Sat, 10 Nov 2007

date.to_formatted_s(:db)            # => "2007-11-10"
date.to_s(:db)                      # => "2007-11-10"

date.to_formatted_s(:short)         # => "10 Nov"
date.to_formatted_s(:number)        # => "20071110"
date.to_formatted_s(:long)          # => "November 10, 2007"
date.to_formatted_s(:long_ordinal)  # => "November 10th, 2007"
date.to_formatted_s(:rfc822)        # => "10 Nov 2007"
date.to_formatted_s(:iso8601)       # => "2007-11-10"

Similarly we can modify to_s ourselves to mimic what functionality we
want it to apply.


On Fri, Jul 31, 2015 at 4:32 PM, Akshat Singhal <akshat@vinsol.com> wrote:

> date is a date object. Not a datetime object.
>
> On Fri, Jul 31, 2015 at 4:30 PM, Jaspreet Singh <notifications@github.com>
> wrote:
>
>> In app/controllers/discussions_controller.rb
>> <https://github.com/SAkshat/iconf/pull/3#discussion_r35964363>:
>>
>> > @@ -48,14 +51,25 @@ def update
>> >
>> >    private
>> >
>> > +    def check_discussion_is_upcoming
>> > +      if !(@discussion.date.to_s > Date.current.to_s)
>>
>> in any case @discussion.date would be a datetime object and so would be
>> Date.current. I don't think you need to convert it to string.
>> Also, we can govern how what to_s does to date objects. Search it and
>> lets discuss.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/SAkshat/iconf/pull/3/files#r35964363>.
>>
>
>
>
> --
>
> Akshat Singhal
> Vinsol
>



-- 

Akshat Singhal
Vinsol